models/resolution/resolution.go

```diff
@@ -166,6 +167,9 @@ func Create(dbp zesty.DBProvider, t *task.Task, resolverInputs map[string]interf
 	if err != nil {
 		return nil, err
 	}
+
+	dst := make([]byte, 0, hex.EncodedLen(len(encryptedSteps)))
```

Comment:
```suggestion
	dst := make([]byte, hex.EncodedLen(len(encryptedSteps)))
```

From the official documentation, it uses the size of the buffer: https://pkg.go.dev/encoding/hex#Encode

Reply:
Note: the functions uses the byte slice indexes, and not append, so the slice must be of the proper length, enough capacity with a 0 length won't work.